src/amp/lib/experiment.ts

```diff
@@ -0,0 +1,150 @@
+interface ExperimentModel<VariantData> {
+    sticky: boolean;
+    consentNotificationId?: string;
+    variants: {
+        [variantName: string]: VariantData;
+    };
+}
+
+type AmpModel = ExperimentModel<number>;
+
+type StyledExperimentModel = ExperimentModel<VariantConfig>;
+
+interface VariantConfig {
+    proportion: number;
+    style: string;
+}
+
+interface ExperimentCollection<ExperimentObject> {
+    [experimentName: string]: ExperimentObject;
+}
+
+export type AmpModelCollection = ExperimentCollection<AmpModel>;
+
+export type StyledModelCollection = ExperimentCollection<StyledExperimentModel>;
+
+type StylesCollection = ExperimentCollection<{ [variantName: string]: string }>;
+
+const extractFromVariants = (
+    variants: { [variantName: string]: VariantConfig },
+    fieldName: keyof VariantConfig,
+): { [variantName: string]: string | number } => {
+    return Object.entries(variants).reduce((acc, [variantName, config]) => {
+        acc = { ...acc, ...{ [variantName]: config[fieldName] } };
+        return acc;
+    }, {});
```

Comment:
- probs not the best idea to reassign the accumulator variable
- no need to build a new inner object with one key&value just to spread it
- needn't have explicit return, can just be an expression

```suggestion
    return Object.entries(variants).reduce((acc, [variantName, config]) => ({
        ...acc, 
        [variantName]: config[fieldName]
    }), {});
```